server/src/api/modules/groups/services/EditGroupService.ts

```diff
@@ -175,19 +182,47 @@ async function executeUpdate(params: EditGroupParams, updatedGroupFields: Prisma
   const joinedEvents: MemberJoinedEvent[] = [];
   const changedRoleEvents: MemberRoleChangeEvent[] = [];
 
+  const nameChanges = await prisma.nameChange.findMany({
+    where: {
+      OR: missingPlayers.map(p => {
+        const equals = p.username;
+        return { newName: { equals, mode: 'insensitive' } };
+      }),
+      status: NameChangeStatus.PENDING
+    },
+    orderBy: { createdAt: 'desc' }
+  });
+
   await prisma
     .$transaction(async transaction => {
-      // Remove any players that are no longer members
-      const removedPlayerIds = await removeExcessMemberships(
+      const excessMemberships = await removeExcessMemberships(
         transaction,
         params.id,
         memberships,
         nextUsernames
       );
 
+      const removeFromAdd: number[] = [];
+      const removeFromExcess: number[] = [];
+
+      for (const nameChange of nameChanges) {
+        const { oldName, newName } = nameChange;
+
+        for (const player of missingPlayers) {
+          const excess = excessMemberships.find(m => m.player.username === oldName.toLowerCase());
```

Comment:
Instead of using .toLowerCase() here (and on L214), i'd use our function standardizeUsername as it not only makes it lower case, but deals with white spacing, trimming, etc.

Also maybe rename this to `match` ? because we're trying to find the "to-remove" membership that matches this nameChange? I'd add some comments explaining what this is